src/Voron/Data/ITree.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Voron.Data
+{
+    public interface ITree
+    {
+        Slice Name { get; }
```

Comment:
Shouldnt be called ```IHasName``` of something like that? Because having an ```ITree``` interface just for debugging/error reporting purposes doesnt make much sense.